### ISSUE TITLE
Regression: Moving resource into new folder breaks scene

### ISSUE THREAD

**Issue by menip** [godotengine/godot#69260]
### Godot version

690199b7d

### System information

Linux Mint

### Issue description

Moving resources such as a png into/out of folders causes existing scenes to lose that resource.

I tested, this behaved as expect in Alpha 14, but didn't bisect further.

Error messages:
```
ERROR: Resource file not found: res://xx.png.
   at: _load (core/io/resource_loader.cpp:219)
ERROR: res://xx.tscn:11 - Parse Error: [ext_resource] referenced non-loaded resource at: res://xx.png
   at: _parse_ext_resource (scene/resources/resource_format_text.cpp:177)
```
### Steps to reproduce

Open repro poject. Move icon.png into top level folder, node_2d.tscn looses it's reference.
Can set Node2D/Icon's image, then move icon.png back into tmp folder. Once again, reference is lost. 

### Minimal reproduction project

[Tmp.tar.gz](https://github.com/godotengine/godot/files/10098665/Tmp.tar.gz)

**Comment by Calinou** [comment]
> I tested, this behaved as expect in Alpha 14, but didn't bisect further.

Do you mean it worked correctly in 4.0.alpha14 but not 4.0.alpha15?

**Comment by nongvantinh** [comment]
Ok, let me handle this.

**Comment by nongvantinh** [comment]
a38891dfb3851d0e119bc418994f9bbf1436be86 is the first bad commit
commit a38891dfb3851d0e119bc418994f9bbf1436be86
Author:  @KoBeWi <kobewi4e@gmail.com>
Date:   Sun Oct 9 13:06:15 2022 +0200

    Fallback to ResourceLoader if can't find UID

Examine a few times, I noticed  when input are `p_path = U"res://test/icon.png"` and `p_generate = false`
the function return `fallback` insead of `ResourceUID::INVALID_ID` like before and the bug happens.

- I've made an effort to comprehend the circumstances surrounding the block code below, but I've failed.
- So far, all I know is that the bug occurs only when we open the scene containing this resource; perhaps just add some checking to see if the scene containing this resource is open or not.
- I need someone who has more authority to give me instructions on it.

https://github.com/godotengine/godot/blob/a38891dfb3851d0e119bc418994f9bbf1436be86/editor/editor_file_system.cpp#L2271-L2274

**Comment by KoBeWi** [comment]
I did a brief testing and:
- this does not happen in another project I tried
- does not happen in MRP if the scene is not currently opened

**Comment by Rindbee** [comment]
v4.0.beta5.official [89a33d28f]

v4.0.beta4.official [e6751549c]

**Comment by lufog** [comment]
@KoBeWi, I didn't notice right away, I thought #69573 completely fixes the problem, but unfortunately it remained to some extent. The scene corrupts if:
- the scene is closed during the renaming of the resource;
- another scene is renamed instead of a resource.

Steps to reproduce (tested on v4.0.beta8.official [45cac42c0]):
- rename `case_1/texture_a.png` while `case_1/scene.tscn` is closed. This will corrupt `case_1/scene.tscn`.
- rename  `case_2/bar.tscn`. This will corrupt `case_2/foo.tscn`.

MRP: [scene_content_corruption_test.zip](https://github.com/godotengine/godot/files/10200393/scene_content_corruption_test.zip)

**Comment by KoBeWi** [comment]
An issue (and a fix) about it already exists: #69794
